### PR TITLE
Expose AdamW parameters

### LINKED ISSUE #1448: Expose betas for AdamW to use with LR Scheduler
I am working on building out my training using Candle. Has been good so far but I missed some of my LR schedulers I was using. Specifically I have been building out OneCycleLR from https://arxiv.org/abs/1708.07120. PyTorch has an implementation [OneCycleLR](https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.OneCycleLR.html#torch.optim.lr_scheduler.OneCycleLR)

In OneCycleLR you modify both the LR and the momentum. For AdamW we can modify the beta1 value for the momentum. Would exposing the betas be invited? I can make a PR but proposing it first.

Here is the scheduler code I have so far [rockerBOO/candle_scheduler](https://github.com/rockerBOO/candle_scheduler). 

Thank you!

### PR DESCRIPTION
**Pull request by rockerBOO**
Wasn't sure how involved this should be but this is the simplest way I thought of. Open for suggestions or improvements. Thank you!

Resolves #1448

### CODE REVIEW

File: candle-nn/src/optim.rs [hunk 1]
```diff
@@ -190,4 +190,12 @@ impl AdamW {
         };
         Self::new(vars, params)
     }
+
+    pub fn params(&self) -> ParamsAdamW {
```
Comment (LaurentMazare) [inline candle-nn/src/optim.rs:194]: What about returning a reference instead so that it's up to the user to decide whether they want to clone it or not?

Comment (rockerBOO) [reply candle-nn/src/optim.rs]: Thank you. Updated.

**Comment by LaurentMazare** [comment]
Thanks!
